addons/stock_picking_batch/models/stock_picking.py

```diff
@@ -47,6 +81,11 @@ class StockPicking(models.Model):
     def create(self, vals_list):
         pickings = super().create(vals_list)
         for picking, vals in zip(pickings, vals_list):
+            if picking.picking_type_id.auto_batch and not vals.get('immediate_transfer') and not vals.get('batch_id') and picking.move_ids:
+                batch_id = picking._find_auto_batch()
+                if batch_id:
+                    vals['batch_id'] = batch_id
```

Comment:
I add the batch rather than the picking in the vals so it triggers the previous condition that runs the sanity check on the created / modified batch.

Reply:
I mean you are sure, you don't want to do picking.batch_id = batch_id? Since when you do pickings = super().create(vals_list) you didn't add 'batch_id' in vals